docs/style-guide.rst

```diff
@@ -214,31 +221,45 @@ No::
 Output Parameters and Return Statements
 
 Yes::
-    
-    function func() public returns (
-        adress someAddressName, 
-        uint256 LongArgument, 
-        uint256 Argument)
+
+    function thisFunctionNameIsReallyLong(
+        address a,
+        address b,
+        address c
+    ) 
+        public 
```

Comment:
I wonder whether it would be more logical to put `public` and `)` at the same indentation level.

Reply:
Currently different indentation is what is recommended in the function declaration section; but, same level indentation seems more logical.
Although changing that would make these our two choices.
```
function thisFunctionNameIsReallyLong(
    address a,
    address b,
    address c
) 
public 
ownerOnly
returns (
    adress someAddressName, 
    uint256 LongArgument, 
    uint256 Argument
)
{
    doSomething()
        
    return (
        veryLongReturnArg1, 
        veryLongReturnArg2, 
        veryLongReturnArg3
    );
}
```

```
function thisFunctionNameIsReallyLong(
    address a,
    address b,
    address c
) 
public 
ownerOnly
returns (
    address someAddressName, 
    uint256 LongArgument, 
    uint256 Argument
) {  
    doSomething()
        
    return (
        veryLongReturnArg1, 
        veryLongReturnArg2, 
        veryLongReturnArg3
    );
} 
```
as opposed to the followng
```
function thisFunctionNameIsReallyLong(
    address a,
    address b,
    address c
) 
    public 
    ownerOnly
    returns (
        address someAddressName, 
        uint256 LongArgument, 
        uint256 Argument
    )
{
    doSomething()
        
    return (
        veryLongReturnArg1, 
        veryLongReturnArg2, 
        veryLongReturnArg3
    );
} 
```

Reply:
I'd go with the first, it is more pragmatic, but admittedly looks worse :)

Reply:
I think the added readability from the first makes it a better choice because the indentation doesn't seem to affect one's understanding of the semantics of the code.  also, this is how the style guide has structured function declarations [since the beginning](https://github.com/ethereum/solidity/blob/64fa9ca478b95970825e3b783275ee66cf848358/docs/style-guide.rst).. we could have an angry mob on our hands if we changed it now lol

Reply:
Actually the current style guide follows a third version:
```
function thisFunctionNameIsReallyLong(
    address a,
    address b,
    address c
) 
    public 
    ownerOnly
    noReentrancy
    returns
(
        address someAddressName, 
        uint256 LongArgument, 
        uint256 Argument
)
{
    doSomething()
        
    return (
        veryLongReturnArg1, 
        veryLongReturnArg2, 
        veryLongReturnArg3
    );
} 
```

Reply:
I wasn't able to find any opinions in regards to multiline output parameters in the [current documentation](http://solidity.readthedocs.io/en/develop/style-guide.html#function-declaration).  When i search for `returns` in that version, I only see cases with single/unnamed parameters.  Can you point me to where you found that other version with the extra indentation after the returns open parenthesis line?

Reply:
@chriseth do you have an opinion on this?
It may be best to just remove the output parameters and return statement from this section anyways, as I they seem to fit better into Function Declarations.
After merge, I can then go ahead and make a PR for this issue and an additional one I wanted to cover in the Function Declarations section.

Reply:
In the most complicated form, you can also have modifiers whose arguments need to be broken into multiple lines.

Please don't see the documentation as authorative on any style matter.

Reply:
@chriseth  Yes, "A foolish consistency is the hobgoblin of little minds".

And, I did remove this area of debate from the current PR, so I think we can merge if there are no other issues.